plugin/web/extensions/sidebar/core/components/desktop/hooks/datasetHooks.ts

```diff
@@ -94,22 +109,30 @@ export default ({
   );
 
   const handleDatasetSave = useCallback(
-    (dataID: string) => {
-      (async () => {
-        if (!inEditor) return;
-        setLoading?.(true);
-        const selectedDataset = project?.datasets.find(d => d.dataID === dataID);
+    async (dataID: string) => {
```

Comment:
Do we not also need to update the prop types where this and the `handleDatasetPublish` are being used?
I was kinda being lazy with types and trying to avoid that by doing the (async()=>{})() style :P
I guess we aren't returning anything in these callbacks so I guess the typings are fine. Is that correct?